private_transactions/src/lib.rs

```
@@ -0,0 +1,782 @@
+		if validator_accounts.is_empty() {
+			self.broadcast_private_transaction(rlp.into());
+			return Ok(());
+		}
+		let private_tx: PrivateTransaction = UntrustedRlp::new(rlp).as_val()?;
+		let contract = private_tx.contract;
+		let contract_validators = self.get_validators(BlockId::Latest, &contract)?;
+
+		match contract_validators
+			.iter()
+			.filter(|&&address| validator_accounts
+				.iter()
+				.any(|&validator| validator == address))
+			.next() {
+			None => {
+				// Not for verification, broadcast further to peers
+				self.broadcast_private_transaction(rlp.into());
+				return Ok(());
+			},
+			Some(&validation_account) => {
+				trace!("Private transaction taken for verification");
+				let original_tx = self.extract_original_transaction(private_tx.clone(), &contract)?;
+				trace!("Validating transaction: {:?}", original_tx);
+				let details_provider = TransactionDetailsProvider::new(&*self.client as &MiningBlockChainClient);
+				let insertion_time = self.client.chain_info().best_block_number;
+				// Verify with the first account available
+				trace!("The following account will be used for verification: {:?}", validation_account);
+				let add_res = self.transactions_for_verification.lock().add_transaction(original_tx, contract, validation_account, private_tx.hash(), &details_provider, insertion_time);
+				match add_res {
+					Ok(_) => {
+						let channel = self.client.get_io_channel();
+						let channel = channel.lock();
+						channel.send(ClientIoMessage::NewPrivateTransaction)
+							.map_err(|_| PrivateTransactionError::ClientIsMalformed.into())
+					},
+					Err(err) => Err(err),
+				}
+			}
+		}
+	}
+
+	/// Private transaction for validation added into queue
+	pub fn on_private_transaction_queued(&self) -> Result<(), PrivateTransactionError> {
+		self.process_queue()
+	}
+
+	/// Retrieve and verify the first available private transaction for every sender
+	fn process_queue(&self) -> Result<(), PrivateTransactionError> {
```

i do not understand why some errors are propagated upstream while the others are (not)silently ignored with traces